library/Icingadb/Common/Macros.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/* Icinga DB Web | (c) 2021 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\Icingadb\Common;
+
+use Icinga\Application\Logger;
+use Icinga\Module\Icingadb\Model\Host;
+use Icinga\Module\Icingadb\Model\Service;
+
+trait Macros
+{
+    /**
+     * Known icinga macros
+     *
+     * @var string[]
+     */
+    protected $icingaMacros = [
+        'HOSTNAME'      => 'name',
+        'HOSTADDRESS'   => 'address',
+        'HOSTADDRESS6'  => 'address6',
+        'host.name'     => 'name',
+        'host.address'  => 'address',
+        'host.address6' => 'address6',
+        'service.name'  => 'name'
```

Comment:
I believe we shouldn't support the all uppercase macros still. These are from Icinga 1.x days and now it's the time to lay them to rest. Anyone running Monitoring in parallel to Icinga DB Web can still use the dot separated macros.

In addition to my comment [here](https://github.com/Icinga/icingadb-web/pull/303#discussion_r703316085) we also have to still allow the differentiation between `host.name` and `service.name`. I hadn't in mind that these were always dot separated macros and only used them as examples. But anyone currently using those together in Monitoring, gets two different results. We also have to support that. So please make this possible, but only for `service.name`. (Consider to just define a path `['service', 'name']` as macro expression and handle arrays differently then)